### PR TITLE
Fix data resolver API

### PR DESCRIPTION
**Pull request by olexii4**
### What does this PR do?
Fix the error for data resolver API 'POST ${baseApiPath}/data/resolver'.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/22814

### Is it tested? How?

1. Deploy Eclipse-Che with the image from this PR.
2. Open Swagger in the new tab.
```{CHE-server}/dashboard/api/swagger```

3. Request data resolver with the next payload:
```json
{
  "url": "{CHE-server}/dashboard/devfile-registry/devfiles/index.json"
}
```
Знімок екрана 2024-02-14 о 14 07 44

The status code should be `200`.

4. Request data resolver with non-existence payload:
```json
{
  "url": "{CHE-server}/dashboard/devfile-registry/devfiles/index111.json"
}
```

Знімок екрана 2024-02-14 о 14 01 33

The status code should be `404`.

### CODE REVIEW

File: packages/dashboard-backend/src/routes/api/helpers/getCertificateAuthority.ts [hunk 1]
```diff
@@ -22,11 +22,15 @@ const certificateAuthority = getCertificateAuthority(
   CHE_SELF_SIGNED_MOUNT_PATH ? CHE_SELF_SIGNED_MOUNT_PATH : DEFAULT_CHE_SELF_SIGNED_MOUNT_PATH,
 );
 
-export const axiosInstance = axios.create({
-  httpsAgent: new https.Agent({
-    ca: certificateAuthority,
-  }),
-});
+export const defaultAxiosInstance = axios.create();
```
Comment (akurinnoy) [inline packages/dashboard-backend/src/routes/api/helpers/getCertificateAuthority.ts:25]:
```suggestion
export const axiosInstanceNoCert = axios.create();
```

Comment (olexii4) [reply packages/dashboard-backend/src/routes/api/helpers/getCertificateAuthority.ts]: I have fixed it.

**Review by akurinnoy — APPROVED** [review]

**Pull request merged**